glusterd2/commands/volumes/volume-reset.go

```diff
@@ -46,6 +46,15 @@ func volumeResetHandler(w http.ResponseWriter, r *http.Request) {
 
 	// Delete the option after checking for volopt flags
 	opReset := false
+
+	if req.All {
+		req.Options = nil
```

Comment:
You don't have to initialise it to nil as you are assigning a new empty slice in the next line.